packages/cozy-stack-client/src/mangoIndex.js

```diff
@@ -82,18 +83,12 @@ export const transformSort = sort => {
  * Compute fields that should be indexed for a mango
  * query to work
  *
- * @param {object} options - Mango query options
- * @param {MangoSelector} [options.selector] - Mango selector
- * @param {MangoPartialFilter} [options.partialFilter] - An optional partial filter
- * @param {MangoSort} [options.sort] - The sorting parameters
  *
  * @returns {Array} - Fields to index
  */
-export const getIndexFields = ({
-  selector = {},
-  partialFilter = {},
-  sort = []
-}) => {
+export const getIndexFields = (
+  /** @type {MangoQueryOptions} */ { selector, partialFilter, sort = [] }
```

Comment:
Shouldn't it be above the function declaration?

Reply:
Nop you can't... 

https://github.com/microsoft/TypeScript/issues/11597
https://github.com/microsoft/TypeScript/pull/17352
https://github.com/microsoft/TypeScript/issues/19645
https://github.com/jsdoc/jsdoc/issues/987#issuecomment-93864964